### DISCUSSION TITLE
how make AJAX call work when using model

### DISCUSSION THREAD

**Discussion by ilia4bit** [django-commons/django-unicorn#355] [Q&A] [answered]
Hey, I'm new to Django Unicorn. it's amazing!
I copy and pasted todo example and its work then I tried to set tasks type to QuerySet[Task]
but when you add a new task you should refresh the page to get updates. here's the code:
```python
class Task(models.Model):
    text = models.TextField()

    def __str__(self):
        return self.text
```
```python
class TodoView(UnicornView):
    task: str = ''
    tasks: QuerySet[Task] = Task.objects.none()

    def mount(self):
        self.tasks = Task.objects.all()

    def add(self):
        Task.objects.create(text=self.task)
        self.task = ''
```
If I add self.mount() to TodoView add() method the component will be updated but is this action OK?

**Comment by adamghill** [comment, marked as answer]
Thanks! I hope it's been easy to work with so far. :) 

Yep, that's a fine way to handle it. `mount` is just another function as far as the actual Python is concerned so it can be called whenever you need to. The other thing I've done in the past to be very explicit about what is happening would be something like:

```python
class TodoView(UnicornView):
    task: str = ''
    tasks: QuerySet[Task] = Task.objects.none()

    def mount(self):
        self.get_all_tasks()

    def get_all_tasks(self):
        self.tasks = Task.objects.all()

    def add(self):
        Task.objects.create(text=self.task)
        self.task = ''

        self.get_all_tasks()
```

A more elaborate example is in the sample project (which attempts to exercise most of the functionality in `Unicorn`): https://github.com/adamghill/django-unicorn/blob/main/example/unicorn/components/models.py. 👍

**Reply by ilia4bit** [reply]
thanks a lot :)
